alicloud/data_source_alicloud_network_interfaces.go

```diff
@@ -65,6 +65,7 @@ func dataSourceAlicloudNetworkInterfaces() *schema.Resource {
 				Type:     schema.TypeList,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				Computed: true,
+				Optional: true,
```

Comment:
There is no need to set `names` to an input parameter and can use `name_regex` instead.